src/utils/path.ts

```diff
@@ -91,7 +91,7 @@ export function parseMetadataXml(fsPath: string): Optional<MetadataXml> {
  * @param directoryName - name of directory to use as a parsing index
  * @returns the FullName
  */
-export function parseNestedFullName(fsPath: string, directoryName: string): Optional<string> {
+export function parseNestedFullName(fsPath: string, directoryName: string): string | undefined {
   const pathSplits = fsPath.split(sep);
```

Comment:
`string | undefined` is the same as `Optional<string>`

Reply:
yes!  

I mildly prefer the explicit, vanilla TS one to the one that requires an import and makes an external contributor or casual github reader go, "hmm...how does that work?".  It does cost 2 characters more 🤪